src/main/kotlin/com/michaelsgroi/baseballreference/BrReports.kt

```diff
@@ -331,21 +325,25 @@ class BrReports(private val brWarDaily: BrWarDaily, private val reportDir: Strin
                 season.toString().padStart(5) + " " +
                 teams.joinToString(",")
 
-    class Report(
+    class Report<T>(
         val name: String,
         val filename: String? = null,
-        val run: () -> List<String>
+        val run: () -> List<T>,
+        val formatter: BrReportFormatter<T>
     )
 
-    private fun buildReport(run: () -> List<String>): Report {
-        return buildReport(emptyList(), run)
+    private fun <T> buildReport(formatter: BrReportFormatter<T>, run: () -> List<T>): Report<T> {
```

Comment:
could be combined into 3rd version (using default arg for `argStrings` (of `emptyList()`)